File: src/test/resources/string_decoder/string_decoder_test.js
load('vertx_tests.js');

function testStringDecoder() {
  var StringDecoder = require('string_decoder').StringDecoder;
  var decoder = new StringDecoder('utf8');
  var cent = new Buffer([0xC2, 0xA2]);
  // TODO: Figure out why this fails
  // vassert.assertEquals("¢", decoder.write(cent));
  vassert.testComplete();
}

initTests(this);

File: src/test/java/org/projectodd/nodej/StringDecoderTest.java
package org.projectodd.nodej;

import static org.fest.assertions.Assertions.assertThat;

import org.junit.Ignore;
import org.junit.Test;

public class StringDecoderTest extends NodejTestSupport {

    @Test
    @Ignore
    public void testWrite() {
        eval("var StringDecoder = require('string_decoder').StringDecoder");
        eval("var decoder = new StringDecoder('utf8')");
        eval("var cent = new Buffer([0xC2, 0xA2])");
        assertThat(eval("decoder.write(cent)")).isEqualTo("¢");
    }
}
